File: src/main/java/models/Booking.java
package models;

import nosqlite.annotations.Document;
import nosqlite.annotations.Id;

@Document
public class Booking {
    @Override
    public String toString() {
        return "Booking{" +
                "id='" + id + '\'' +
                ", userId='" + userId + '\'' +
                ", leaseId='" + leaseId + '\'' +
                ", location='" + location + '\'' +
                ", startDate='" + startDate + '\'' +
                ", endDate='" + endDate + '\'' +
                ", guests=" + guests +
                ", totalPrice=" + totalPrice +
                '}';
    }

    @Id
    private String id;
    private String userId;
    private String leaseId;
    private String location;
    private String startDate;
    private String endDate;
    private int guests;
    private double totalPrice;

    public Booking() {}

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getLeaseId() {
        return leaseId;
    }

    public void setLeaseId(String leaseId) {
        this.leaseId = leaseId;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public String getStartDate() {
        return startDate;
    }

    public void setStartDate(String startDate) {
        this.startDate = startDate;
    }

    public String getEndDate() {
        return endDate;
    }

    public void setEndDate(String endDate) {
        this.endDate = endDate;
    }

    public int getGuests() {
        return guests;
    }

    public void setGuests(int guests) {
        this.guests = guests;
    }

    public double getTotalPrice() {
        return totalPrice;
    }

    public void setTotalPrice(double totalPrice) {
        this.totalPrice = totalPrice;
    }

    @Override
    public String toString() {
        return "Booking{" +
                "id='" + id + '\'' +
                ", userId='" + userId + '\'' +
                ", leaseId='" + leaseId + '\'' +
                ", location='" + location + '\'' +
                ", startDate='" + startDate + '\'' +
                ", endDate='" + endDate + '\'' +
                ", guests=" + guests +
                ", totalPrice=" + totalPrice +
                '}';
    }
}

File: frontend/src/components/Booking.js
export default class Booking {
  constructor(
    userId,
    leaseId,
    location,
    startDate,
    endDate,
    guests,
    totalPrice,
    bookedStay
  ) {
    this.userId = userId;
    this.leaseId = leaseId;
    this.location = location;
    this.startDate = startDate;
    this.endDate = endDate;
    this.guests = guests;
    this.totalPrice = totalPrice;
    this.bookedStay = bookedStay;
  }

  getUserId() {
    return this.userId;
  }
  setUserId(newUserId) {
    this.userId = newUserId;
  }
  getLeaseId() {
    return this.leaseId;
  }
  setLeaseId(newLeaseId) {
    this.leaseId = newLeaseId;
  }
  getLocation() {
    return this.location;
  }
  setLocation(newLocation) {
    this.location = newLocation;
  }
  getStartDate() {
    return this.startDate;
  }
  setStartDate(newStartDate) {
    this.startDate = newStartDate;
  }
  getEndDate() {
    return this.endDate;
  }
  setEndDate(newEndDate) {
    this.endDate = newEndDate;
  }
  getGuests() {
    return this.guests;
  }
  setGuests(newAmountOfGuests) {
    this.guests = newAmountOfGuests;
  }
  getTotalPrice() {
    return this.totalPrice;
  }
  setTotalPrice(newTotalPrice) {
    this.totalPrice = newTotalPrice;
  }
  getBookedStay() {
    return this.bookedStay;
  }
  setBookedStay(newBookedStay) {
    this.bookedStay = newBookedStay;
  }
  isInThePast(toCompare) {
    let correctFormatToCompare = this.getDateInCorrectFormat(toCompare);
    let splitFirstDate = correctFormatToCompare.split('-');
    let todayInCorrectFormat = this.getDateInCorrectFormat(new Date());
    let splitSecondDate = todayInCorrectFormat.split('-');
    if (
      splitFirstDate[0] <= splitSecondDate[0] &&
      splitFirstDate[1] <= splitSecondDate[1] &&
      splitFirstDate[2] <= splitSecondDate[2]
    ) {
      return true;
    }
    return false;
  }
  getDateInCorrectFormat(myDate) {
    let today = new Date(myDate);
    let month = today.getMonth() + 1;
    let day = today.getDate();
    let zeroBeforeMonth = '';
    let zeroBeforeDay = '';
    if (month < 10) {
      zeroBeforeMonth += '0' + month;
      month = zeroBeforeMonth;
    }
    if (day < 10) {
      zeroBeforeDay += '0' + day;
      day = zeroBeforeDay;
    }
    return today.getFullYear() + '-' + month + '-' + day;
  }
}
